### ISSUE TITLE
Make `m_CanCrossLanes` work intelligently

### ISSUE THREAD

**Issue by aubergine10** [CitiesSkylinesMods/TMPE#646]
This is possibly multiple separate issues but they all seem to link with the (non-funcitonal) "Can cross lanes" setting in asset editor. I'll try and describe as best I can.

In the asset editor, there's a property "Can Cross Lanes". All the indicators are that it was intended prevent vehicles cutting across multiple lanes of oncoming traffic on medium or large roads. It looks like it was something that CO were mid way through implementing but dropped prior to initial release of the game back in 2015.

Asset creators making road and track networks have desired the feature to work for long time, however it would probably need to behave differently for roads vs. tracks (and ideally have manual override + customisable prefab default for end-users).

Users currently have to use lane connectors to force correct behaviour. However, it would be nice if the "Can cross lanes" flag could be used by asset authors to automatically set applicable road and rail junctions to the correct behaviour.

## Tracks

MOM and, soon, RWY (Railway), contain 4-track rail/metro networks, for example:

![image](https://user-images.githubusercontent.com/1386719/73690029-6d098c80-46c7-11ea-8915-fe4aac0e923a.png)

The 4 track networks are useful for:

1. [Bypass stations](https://steamcommunity.com/workshop/filedetails/?id=1577947171) - this is the main use
2. Higher capacity rail netorks without the hassle of trying to align  2 x 2-track networks side by side (eg. allowing passenger and cargo trains to roll side by side). This is a side use that would require something akin to #634 at one of the nodes/segments elsewhere on the track to allow trains to diverge and merge in order to swap between outer and inner track in the direction they are going.

As you can imagine, the game struggles with these configurations.

The tracks usually have 2 lanes in one direction then 2 lanes in the other direction as indicated on image above. If `m_CanCrossLanes == false`, it should ideally give effect of the lane connectors on image above when the 4-lane track splits in to 2 x 2-lane track.

It seems there was some attempt to deal with 'ahead only at junctions' in TMPE 10.17 - see: #230 and #236. It was raised again in the ill-fated #243 but fizzled out.

For "bypass station tracks" (see link above) there is mixture of (outer) station tracks and (inner) rail tracks that also needs accommodating. The station tracks are one-lane one-sided station segments, and then there's normal middle double track; trains on the central tracks can just bypass (ie. drive striaght through without stopping) the station.

Should definitely reach out to MOM team as there is likely something from TMPE 10.17 that I assume works to some extent; either that or they patch TMPE somehow? BadPeanut (can be found in the /r/CitiesSkylines discord) will almost certainly know how the MOM tracks work.

REV0 will be providing WIP versions of RWY quad tracks for testing.

## Roads

For roads, `m_CanCrossLanes == false` would be perfect for this:

![image](https://user-images.githubusercontent.com/1386719/73690683-d1791b80-46c8-11ea-9462-90b71a5c3730.png)

In other words, don't cross oncoming traffic at a junction.

But currently users have to do this kind of thing:

![image](https://user-images.githubusercontent.com/1386719/73690641-b8706a80-46c8-11ea-867e-ad08ef7a0614.png)

It's touched upon in #89, #243 and also related to #503 and #504.

**Comment by aubergine10** [comment]
Related:  #558

**Comment by kianzarrin** [comment]
@aubergine10 can we close this issue? there are so many duplicates of this issue that I am getting confused!

Instead of telling TMPE to stop cars from crossing the median we can look at this the other way around: hide he median if cars can cross.

My DCR mod already does this: https://steamcommunity.com/sharedfiles/filedetails/?id=2104976832
- hides broken medians (this is the part that solves this issue)
- generates DC meshes (optional)

So can we close this and all the node median ones? (#89 cannot be closed because it is in the middle of the segment and is a different matter).

**Comment by kianzarrin** [comment]
@aubergine10 The track part of the issue is still valid
![image](https://user-images.githubusercontent.com/26344691/145279320-cef8a930-1b39-459b-bf1f-34815b9a405b.png)

FYI my Adaptive Networks mod will have a track feature that can totally hide all unconnected tracks. 
https://steamcommunity.com/sharedfiles/filedetails/?id=2669938594
![image](https://user-images.githubusercontent.com/26344691/145280322-6f35406a-0e6c-436e-848f-6865296069e8.png)

**Comment by aubergine10** [comment]
I think we could probably close a lot of these, if the various issues are summarised in to a central issue maybe?

I'll have a go at creating that new issue, listing all the spammy issues, and then, if everyone agrees, I'll close the spammy issues.

**Comment by aubergine10** [comment]
Closing this as Kian's mods are handling it. (That AN mod doing the custom node meshes is freaking awesome!!)

For other stuff relating to medians and crossing roads/lanes, see #1223
